files/unarchive.py

```diff
@@ -347,7 +347,7 @@ def is_unarchived(self):
             version = pcs[1]
             ostype = pcs[2]
             size = int(pcs[3])
-            path = pcs[7]
+            path = unicode(pcs[7], 'utf-8')
```

Comment:
unicode is not available on python3.  This also won't handle non-utf8 filenames but I'm not sure we care about those for now.  Do you remember if any of the old bugs were wrt non-utf8 filenames inside of the tar or zip?

proposed code to deal with the python3 issue:
``` python
from ansible.module_utils._text import to_text
[...]
path = to_text(pcs[7], errors='surrogate_or_strict')
```